diff --git a/configure.ac b/configure.ac
index faa0379..a7e0c45 100644
--- a/configure.ac
+++ b/configure.ac
@@ -358,7 +358,7 @@ AS_IF([test "x$GLDISPATCH_PAGE_SIZE" != "x"],
 
 
 dnl default CFLAGS
-CFLAGS="$CFLAGS -Wall -Werror -include config.h -fvisibility=hidden $DEFINES"
+CFLAGS="$CFLAGS -Wall -include config.h -fvisibility=hidden $DEFINES"
 
 AC_CONFIG_FILES([Makefile
                  libglvnd.pc
