diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4967b94..3478f1b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -160,10 +160,10 @@ if ("${CMAKE_C_COMPILER_ID}" STREQUAL "MSVC")
     endif()
 endif()
 
-add_subdirectory(examples)
+# add_subdirectory(examples)
 add_subdirectory(src)
 add_subdirectory(tests)
-add_subdirectory(tools)
+# add_subdirectory(tools)
 
 if (K4A_BUILD_DOCS)
     find_package(Doxygen 1.8.14 EXACT)
@@ -201,9 +201,10 @@ option(K4A_VALIDATE_CLANG_FORMAT "Validate clang-format results as part of build
 set (CLANG_FORMAT_ROOT
     ${CMAKE_CURRENT_SOURCE_DIR}/src
     ${CMAKE_CURRENT_SOURCE_DIR}/include
-    ${CMAKE_CURRENT_SOURCE_DIR}/examples
+    # ${CMAKE_CURRENT_SOURCE_DIR}/examples
     ${CMAKE_CURRENT_SOURCE_DIR}/tests
-    ${CMAKE_CURRENT_SOURCE_DIR}/tools)
+    # ${CMAKE_CURRENT_SOURCE_DIR}/tools
+    )
 
 find_program(CLANG_FORMAT clang-format DOC "Clang format tool")
 if (${CLANG_FORMAT} STREQUAL "CLANG_FORMAT-NOTFOUND")
@@ -318,12 +319,12 @@ cpack_add_component(
     DEPENDS
         runtime)
 
-cpack_add_component(
-    tools
-    DISPLAY_NAME
-        Tools
-    DESCRIPTION
-        "Tools for Azure Kinect Development"
-    REQUIRED
-    DEPENDS
-        runtime)
+# cpack_add_component(
+#     tools
+#     DISPLAY_NAME
+#         Tools
+#     DESCRIPTION
+#         "Tools for Azure Kinect Development"
+#     REQUIRED
+#     DEPENDS
+#         runtime)
