diff --git a/bin/aclocal.in b/bin/aclocal.in
index b3715d9..f716982 100644
--- a/bin/aclocal.in
+++ b/bin/aclocal.in
@@ -68,8 +68,8 @@ $perl_threads = 0;
 # ACLOCAL_PATH environment variable, and reset with the '--system-acdir'
 # option.
 my @user_includes = ();
-my @automake_includes = ('@datadir@/aclocal-' . $APIVERSION);
-my @system_includes = ('@datadir@/aclocal');
+my @automake_includes = split(':',$ENV{"ACLOCAL_PATH"});
+my @system_includes = ($ENV{"ACLOCAL_M4_PATH"});
 
 # Whether we should copy M4 file in $user_includes[0].
 my $install = 0;
