diff --git a/CMakeLists.txt b/CMakeLists.txt
index 78834fe1..ad492fd7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -20,6 +20,20 @@ os_set_flags()
 # macro definition located at "CMake/global_config.cmake"
 global_set_flags()
 
+find_package(PkgConfig)
+if(NOT PKG_CONFIG_FOUND)
+    message(FATAL_ERROR "\n\n PkgConfig package is missing!\n\n")
+endif()
+
+pkg_search_module(LIBUSB1 REQUIRED libusb-1.0)
+if(LIBUSB1_FOUND)
+    include_directories(SYSTEM ${LIBUSB1_INCLUDE_DIRS})
+    link_directories(${LIBUSB1_LIBRARY_DIRS})
+    list(APPEND librealsense_PKG_DEPS "libusb-1.0")
+else()
+    message( FATAL_ERROR "Failed to find libusb-1.0" )
+endif(LIBUSB1_FOUND)
+
 if(BUILD_SHARED_LIBS)
     add_library(${LRS_TARGET} SHARED "")
 else()
