diff --git a/gst/mpegtsdemux/tsdemux.c b/gst/mpegtsdemux/tsdemux.c
index 64f9d0bb7..2cc2549b4 100644
--- a/gst/mpegtsdemux/tsdemux.c
+++ b/gst/mpegtsdemux/tsdemux.c
@@ -73,7 +73,7 @@
 #define GST_FLOW_REWINDING GST_FLOW_CUSTOM_ERROR

 /* latency in nsecs */
-#define TS_LATENCY (700 * GST_MSECOND)
+#define TS_LATENCY (50 * GST_MSECOND)

 GST_DEBUG_CATEGORY_STATIC (ts_demux_debug);
 #define GST_CAT_DEFAULT ts_demux_debug
