diff --git a/projects/clang/lib/Driver/ToolChains/Gnu.cpp b/projects/clang/lib/Driver/ToolChains/Gnu.cpp
index c8a7fce..d03f297 100644
--- a/projects/clang/lib/Driver/ToolChains/Gnu.cpp
+++ b/projects/clang/lib/Driver/ToolChains/Gnu.cpp
@@ -2860,6 +2860,11 @@ void Generic_GCC::AddClangCXXStdlibIncludeArgs(const ArgList &DriverArgs,
       DriverArgs.hasArg(options::OPT_nostdincxx))
     return;
 
+  if (::getenv("LIBCXX_INCLUDE_PATH")) {
+    addSystemInclude(DriverArgs, CC1Args, ::getenv("LIBCXX_INCLUDE_PATH"));
+    return;
+  }
+
   switch (GetCXXStdlibType(DriverArgs)) {
   case ToolChain::CST_Libcxx:
     addLibCxxIncludePaths(DriverArgs, CC1Args);
