diff --git a/CMakeLists.txt b/CMakeLists.txt
index c37c90c..95d179e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -54,6 +54,9 @@ endmacro()
 #add_subdirectory(src)
 aux_source_directory(src src_MAIN)
 if(AMQP-CPP_LINUX_TCP)
+    find_package(PkgConfig REQUIRED)
+    pkg_check_modules(OPENSSL REQUIRED openssl)
+    include_directories(SYSTEM ${CMAKE_CURRENT_SOURCE_DIR}/include ${OPENSSL_INCLUDE_DIRS})
     #add_subdirectory(src/linux_tcp)
     aux_source_directory(src/linux_tcp src_LINUX_TCP)
 endif()
@@ -128,5 +131,5 @@ target_include_directories(${PROJECT_NAME} SYSTEM PUBLIC
 )
 
 if(AMQP-CPP_LINUX_TCP)
-    target_link_libraries(${PROJECT_NAME} ${CMAKE_DL_LIBS})
+    target_link_libraries(${PROJECT_NAME} ${CMAKE_DL_LIBS} ${OPENSSL_LIBRARIES} )
 endif()
